parts/k8s/kubernetescustomscript.sh

```diff
@@ -1,5 +1,17 @@
 #!/bin/bash
 
+# This script runs on every Kubernetes VM
+# Exit codes represent the following:
+# | exit code number | meaning |
```

Comment:
How about if we reserve 1-9 for "general" exit failures, and then a decimal range (e.g., 10-19) per application-specific error condition. So, for example, "Etcd data dir not found" would be 10; "Timeout waiting for etcd to be accessible" would be 11; and then "Timeout waiting for docker install to finish" would be 20; etc

Reply:
Sorry, 1 and 2 should be reserved:

http://www.tldp.org/LDP/abs/html/exitcodes.html

Reply:
So, 3-9 for general errors:

- a service could not be enabled by systemctl
- a service could not be started by systemctl
- timeout waiting for cloud-init runcmd to complete
- timeout waiting for a file (ideally we'd have separate error messages per file we are waiting for, as they indicate a particular serial dependency)

Reply:
updated them, is it what you had in mind?